File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        low = float('inf')

        for i in prices:
            if i<low:
                low = i
            profit = i-low
            if profit > max_profit:
                max_profit = profit
        return max_profit 

       
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int low = prices[0];
        int profit = 0;
        
        for(int i:prices){
            if(i<low){
                low = i;
            }
            profit = i-low;
            max_profit = max(max_profit,profit);
        }
        return max_profit;
    }
};